File: api/src/main/java/com/wbrawner/twigs/session/UserSessionRepository.java
package com.wbrawner.twigs.session;

import org.springframework.data.repository.PagingAndSortingRepository;

import java.util.Date;
import java.util.List;
import java.util.Optional;

public interface UserSessionRepository extends PagingAndSortingRepository<Session, String> {
    List<Session> findByUserId(String userId);

    Optional<Session> findByToken(String token);

    Optional<Session> findByUserIdAndToken(String userId, String token);

    void deleteAllByExpirationBefore(Date expiration);
}

File: api/src/main/kotlin/com/wbrawner/budgetserver/session/UserSessionRepository.kt
package com.wbrawner.budgetserver.session

import org.springframework.data.repository.PagingAndSortingRepository
import java.util.*

interface UserSessionRepository : PagingAndSortingRepository<Session, String> {
    fun findByUserId(userId: String?): List<Session?>?
    fun findByToken(token: String?): Optional<Session?>?
    fun findByUserIdAndToken(userId: String?, token: String?): Optional<Session?>?
    fun deleteAllByExpirationBefore(expiration: Date?)
}